src/test/java/org/junit/tests/experimental/AssumptionTest.java

```diff
@@ -94,10 +99,15 @@ public void assumeThatPassesOnStrings() {
         assertCompletesNormally();
     }
 
-    @Test(expected = AssumptionViolatedException.class)
+    @Test
     public void assumeNotNullThrowsException() {
-        Object[] objects = {1, 2, null};
-        assumeNotNull(objects);
+        try {
+            Object[] objects = {1, 2, null};
```

Comment:
True.
+1

Il 13/mag/2016 14:56, "Kevin Cooney" <notifications@github.com> ha scritto:
>
> In src/test/java/org/junit/tests/experimental/AssumptionTest.java:
>
> >      public void assumeNotNullThrowsException() {
> > -        Object[] objects = {1, 2, null};
> > -        assumeNotNull(objects);
> > +        try {
> > +            Object[] objects = {1, 2, null};
>
> I am personally not a fan of using the "expected" attribute of @Test. It
doesn't allow you to specify which line should throw the exception, and as
you've seen it can have bugs (the one you found wasn't the first)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub